File: src/main/java/com/tylersadler/controller/MemberController.java
package com.tylersadler.controller;

import com.tylersadler.dto.MemberDto;
import com.tylersadler.service.MemberService;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.inject.Inject;
import java.util.List;

@RestController
@RequestMapping("/api")
public class MemberController {

    private final MemberService memberService;

    @Inject
    public MemberController(MemberService memberService){
        this.memberService = memberService;
    }

    @RequestMapping(value = "/members", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public List<MemberDto> findAll(){
        return memberService.findAll();
    }
}

File: src/main/webapp/directive/member/memberController.js
(function (angular) {
    'use strict';
    angular.module('gymApp').controller('MemberController', memberController);

    memberController.$inject = ['$scope', 'MemberService'];

    function memberController($scope, MemberService) {
        $scope.findAllMembers = function () {
            MemberService.findAll().then(function (members) {
                $scope.members = members;
            });
        };

        $scope.findAllMembers();
    }
})(angular);